protos/flyteidl/service/dataproxy.proto

```diff
@@ -102,6 +104,29 @@ message CreateDownloadLinkResponse {
   google.protobuf.Timestamp expires_at = 2;
 }
 
+// Wraps a flyte url, a unique identifier in the form of flyte://<something> that uniquely, for a given Flyte
+// backend, identifies a Flyte artifact (input, output, flyte deck, etc.).
+// e.g. flyte://v1/proj/development/execid/n2/0/i (for 0th task execution attempt)
+//      flyte://v1/proj/development/execid/n2/i (for node execution)
+message FlyteArtifact {
+  string flyte_url = 1;
+}
+
+// General request artifact to retrieve data from a Flyte artifact url.
+message GetDataRequest {
+  FlyteArtifact artifact = 1;
+}
+
+message GetDataResponse {
+  oneof data {
+    // literal map data will be returned
+    core.LiteralMap literal_map                = 1;
+
+    // Flyte deck html will be returned as a signed url users can download
+    CreateDownloadLinkResponse flyte_deck_download_link = 2;
```

Comment:
embedding one response in another seems like a code-smell

Reply:
oh man i forget did we end up deciding on using dataproxy? if so i guess +1. Also response inside response is really not loved. it confuses the client user, but i see there is no wrapper inside it. 

By the way should we also add link to the checkpoint if it is available - like in the task node case - https://github.com/flyteorg/flyteidl/blob/c7ae927f73df9ecdbbf01b6021836b9b2079e527/protos/flyteidl/event/event.proto#L128? I do not think it should be signed.

Reply:
I think a separate object sounds better...